src/manage_nsfs/manage_nsfs_validations.js

```diff
@@ -738,6 +740,39 @@ function validate_connection_args(user_input, action) {
     }
 }
 
+////////////////////////////
+//// UPGRADE VALIDATION ////
+///////////////////////////
+
+/**
+ * validate_expected_version checks that expected_version is valid
+ * if it's not valid it throws an error
+ * @param {string} expected_version
+ */
+function validate_expected_version(expected_version) {
+    if (!expected_version) {
+        throw new Error('expected_version flag is required');
+    }
+    if (!version_utils.is_valid_sematic_version(expected_version)) {
+        throw new Error('expected_version must have sematic version structure (major.minor.patch)');
+    }
+    if (check_if_expected_version_newer_that_package_version(expected_version)) {
```

Comment:
method name is confusing, It gives a feeling that an error is thrown when the expected version is latest.
And I am not sure if a long method name like this is really helpful or not.

Reply:
I tried to change it, if you think you have a better suggestion please suggest.